src/lesson4/Calculator.java

```diff
@@ -0,0 +1,180 @@
+package lesson4;
+
+import java.util.Scanner;
+
+public class Calculator {
+    private double currentValue = 0;
+
+    public Calculator() {}
+
+    public Calculator(double currentValue) {
+        this.currentValue = currentValue;
+    }
+
+    public double sum(double num1, double num2) {
+        currentValue = num1 + num2;
+        return currentValue;
+    }
+
+    public double subtraction(double num1, double num2) {
+        currentValue = num1 - num2;
+        return currentValue;
+    }
+
+    public double multiplication(double num1, double num2) {
+        currentValue = num1 * num2;
+        return currentValue;
+    }
+
+    public double division(double num1, double num2) {
+        currentValue = num1 / num2;
+        return currentValue;
+    }
+
+    public void commandC() {
+        currentValue = 0;
+        calculatorLook(currentValue);
+    }
+
+    public void commandBackspace() { //only for currentValue
+        double number = currentValue;
+        String numberStr = Double.toString(number);
+        int i = 0;
+        while (numberStr.toCharArray()[numberStr.length() - 1] == '.' | numberStr.toCharArray()[numberStr.length() - 1] == '0') {
```

Comment:
complicated boolean operation, try to make it smaller or use different loop

Reply:
i cant find out, how should i do this

Reply:
you have int var I, why you don't try to use for() loop ? try to rewrite logic with while{} to for{} loop, if you did not succeed we can discus it on Friday